source/common/http/conn_manager_impl.cc

```diff
@@ -905,8 +905,9 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
       request_headers_->Expect()->value() == Headers::get().ExpectValues._100Continue.c_str()) {
     // Note in the case Envoy is handling 100-Continue complexity, it skips the filter chain
     // and sends the 100-Continue directly to the encoder.
-    chargeStats(continueHeader());
-    response_encoder_->encode100ContinueHeaders(continueHeader());
+    const auto& continue_headers = continueHeader();
+    ASSERT(response_encoder_->encode100ContinueHeaders(continue_headers).ok());
+    chargeStats(continue_headers);
```

Comment:
`chargeStats` assumes that the required headers are present, so we need to change the order here and elsewhere.